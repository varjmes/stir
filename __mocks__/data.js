export const data = {
  "site": {
    "siteMetadata": {
      "title": "Test title",
      "description": "Test description",
      "author": "Test author"
    }
  }
}

export const indexData = {
  "site": {
    "siteMetadata": {
      "title": "Test title",
      "description": "Test description",
      "author": "Test author"
    }
  },
  "allMarkdownRemark": {
    "totalCount": 2,
    "edges": [
      {
        "node": {
          "id": "123",
          "frontmatter": {
            "title": "Test Post",
            "date": "12 March, 2019",
            "author": "Test author",
            "description": "Test description",
            "keywords": ["test", "test1", "test2"]
          },
          "excerpt": "Test excerpt",
          "fields": {
            "slug": "/test/1/"
          }
        }
      },
      {
        "node": {
          "id": "456",
          "frontmatter": {
            "title": "Test Post",
            "date": "11 March, 2019"
          },
          "excerpt": "Test excerpt",
          "fields": {
            "slug": "/test/2"
          }
        }
      }
    ]
  }
}

export const recipeData = {
  "site": {
    "siteMetadata": {
      "title": "Test Recipe",
      "description": "Test Recipe description",
      "author": "Test recipe author"
    }
  },
  "markdownRemark": {
    "html": "<p>Wow, eggs.</p>",
    "frontmatter": {
      "title": "Eggs"
    }
  }
}
