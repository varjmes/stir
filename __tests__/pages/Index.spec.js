import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery, StaticQuery } from 'gatsby'

import Index from '../../src/pages/index'
import { indexData } from '../../__mocks__/data'

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) => render(indexData)),
    useStaticQuery.mockReturnValue(indexData)
})

describe('Index page', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Index data={indexData} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
