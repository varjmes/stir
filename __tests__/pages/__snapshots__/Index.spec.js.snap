// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page renders correctly 1`] = `
<h1
  className="sc-bdVaJa LoiiH"
>
  stir.
</h1>
`;
